dosbox: FTBFS on aarch64-darwin

### Steps To Reproduce
Steps to reproduce the behavior:
1. build `dosbox` on aarch64-darwin

### Build log
https://hydra.nixos.org/build/190016824/log

### Additional context
Build stopped working some time between [2022-07-02](https://hydra.nixos.org/build/182487544/log) and 2022-09-10 (the above). Notable changes in the build output:
```
[autoreconfHook is executed]
[SDL_compat instead of real SDL]
[alot of type sizes are detected as 0:]
checking size of unsigned char... 0
checking size of unsigned short... 0
checking size of unsigned int... 0
checking size of unsigned long... 0
checking size of unsigned long long... 0
...
[SDL_net not correctly detected anymore:]
checking for SDL_net.h... no
checking for SDLNet_Init in -lSDL_net... yes
configure: WARNING: Can't find SDL_net, internal modem and ipx disabled
...
[SDL_sound not correctly detected anymore:]
checking for SDL_sound.h... no
checking for Sound_Init in -lSDL_sound... yes
checking for Sound_Seek in -lSDL_sound... yes
configure: WARNING: Can't find libSDL_sound, libSDL_sound support disabled
...
[borked:]
clang++ -DHAVE_CONFIG_H -I. -I../..  -I../../include -I/nix/store/sf8pqf0d4zp474xwi5qd2ckvym2jrfrp-SDL_compat-1.2.52/include/SDL -D_THREAD_SAFE  -g -O2 -mno-ms-bitfields  -c -o callback.o callback.cpp
In file included from callback.cpp:24:
In file included from ../../include/dosbox.h:24:
../../config.h:275:4: error: "sizeof (unsigned char) != 1"
#  error "sizeof (unsigned char) != 1"
   ^
../../config.h:282:4: error: "sizeof (unsigned short) != 2"
#  error "sizeof (unsigned short) != 2"
   ^
../../config.h:295:4: error: "can't find sizeof(type) of 4 bytes!"
#  error "can't find sizeof(type) of 4 bytes!"
   ^
../../config.h:305:4: error: "can't find data type of 8 bytes"
#  error "can't find data type of 8 bytes"
   ^
../../config.h:312:11: error: unknown type name 'Bit64u'
  typedef Bit64u Bitu;
          ^
../../config.h:313:11: error: unknown type name 'Bit64s'
  typedef Bit64s Bits;
          ^
[many more errors follow]
```

I sadly lack the hardware to do any testing on this, just noticed it while trying to let CI test a new platform support of something that depends on this: https://github.com/NixOS/nixpkgs/pull/203943#issuecomment-1334119478

### Notify maintainers
@matthewbauer (maintainer of `dosbox`)
@peterhoeg (merged last change to `dosbox`, can't see an ofborg run on aarch64-darwin in the checks)

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

Whatever Hydra is using.

--- a/pkgs/applications/emulators/dosbox/default.nix
+++ b/pkgs/applications/emulators/dosbox/default.nix
@@ -2,7 +2,7 @@
 , stdenv
 , fetchurl
 , autoreconfHook
-, SDL_compat
+, SDL
 , SDL_net
 , SDL_sound
 , copyDesktopItems
@@ -29,7 +29,7 @@ stdenv.mkDerivation rec {
   ];
 
   buildInputs = [
-    SDL_compat
+    SDL
     SDL_net
     SDL_sound
     libGL

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -1961,7 +1961,9 @@ with pkgs;
 
   dlx = callPackage ../applications/emulators/dlx { };
 
-  dosbox = callPackage ../applications/emulators/dosbox { };
+  dosbox = callPackage ../applications/emulators/dosbox {
+    SDL = if stdenv.isDarwin then SDL else SDL_compat;
+  };
 
   dosbox-staging = callPackage ../applications/emulators/dosbox-staging { };
 

